modyn/selector/internal/selector_strategies/abstract_selection_strategy.py

```diff
@@ -106,6 +99,12 @@ def __init__(
             )
 
         self._trigger_sample_directory = self._modyn_config["selector"]["trigger_sample_directory"]
+        self._storage_backend = self._init_storage_backend()
+
+    @abstractmethod
+    def _init_storage_backend(self) -> AbstractStorageBackend:
+        """Initializes the storage backend."""
+        raise NotImplementedError
```

Comment:
I found every selection strategy has a `StorageBackend`, it is just because we didn't implement both backends for all strategies, that the `self_storage_backend` is not able to become a field in the superclass.

This way it will become a field in the superclass. This has the additional advantage that the code to decide if we do multi-threading, i.e.
```
self._insertion_threads = modyn_config["selector"]["insertion_threads"]
self._is_test = "PYTEST_CURRENT_TEST" in os.environ
self._is_mac = platform.system() == "Darwin"
self._disable_mt = self._insertion_threads <= 0
```
previously redundant in both selection strategy and storage backend, can be deduplicated.